File: leetcode/python/867_transpose_matrix.py
class Solution:
    def transpose(self, matrix: list[list[int]]) -> list[list[int]]:
        mat: list[list[int]] = [[0] * len(matrix) for _ in range(len(matrix[0]))]
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                mat[j][i] = matrix[i][j]
        return mat
        
s = Solution()
print(s.transpose([[1,2,3],[4,5,6],[7,8,9]]))
print(s.transpose([[1,2,3],[4,5,6]]))

File: leetcode/rust/src/867_transpose_matrix.rs
impl Solution {
    // 1  2  3 
    // 4. 5. 6

    // 1 4
    // 2 5
    // 3 6 
    pub fn transpose(matrix: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let m = matrix.len();
        let n = matrix[0].len();
        let mut res = vec![vec![0; m]; n];

        for i in 0..m {
            for j in 0..n {
                res[j][i] = matrix[i][j];
            }
        }
        res
    }
}